packages/rulesets/src/shared/functions/uniquenessTags.ts

```diff
@@ -0,0 +1,61 @@
+import { createRulesetFunction } from '@stoplight/spectral-core';
+
+import type { IFunctionResult } from '@stoplight/spectral-core';
+
+type Tags = Array<{ name: string }>;
+
+function getDuplicateTagNames(tags: Tags): string[] {
+  const tagNames = tags.map(item => item.name);
+  return tagNames.reduce((acc, item, idx, arr) => {
+    if (arr.indexOf(item) !== idx && acc.indexOf(item) < 0) {
+      acc.push(item);
+    }
+    return acc;
+  }, [] as string[]);
+}
+
+export default createRulesetFunction<Tags, null>(
+  {
+    input: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          name: {
+            type: 'string',
+          },
+        },
+      },
+    },
+    options: null,
+  },
+  function uniquenessTags(targetVal, _, ctx) {
+    const duplicatedTags = getDuplicateTagNames(targetVal);
+    if (duplicatedTags.length === 0) return [];
+
+    const results: IFunctionResult[] = [];
+
+    duplicatedTags.map(duplicatedTag => {
+      let checkedFirst = false;
+      const duplicatedTags: number[] = [];
+      targetVal.forEach((tag, index) => {
+        if (tag.name === duplicatedTag) {
+          if (!checkedFirst) {
+            checkedFirst = true;
+            return;
+          }
+          duplicatedTags.push(index);
+        }
+      });
+
+      results.push(
+        ...duplicatedTags.map(duplicatedIndex => ({
+          message: `"tags" object contains duplicate tag name "${duplicatedTag}".`,
+          path: [...ctx.path, duplicatedIndex, 'name'],
+        })),
+      );
+    });
+
+    return results;
+  },
+);
```

Comment:
How about this?
```suggestion
import { createRulesetFunction, IFunctionResult } from '@stoplight/spectral-core';

type Tags = Array<{ name: string }>;

function getDuplicateTags(tags: Tags): number[] {
  return tags
    .map(item => item.name)
    .reduce<number[]>((acc, item, i, arr) => {
      if (arr.indexOf(item) !== i) {
        acc.push(i);
      }

      return acc;
    }, []);
}

export default createRulesetFunction<Tags, null>(
  {
    input: {
      type: 'array',
      items: {
        type: 'object',
        properties: {
          name: {
            type: 'string',
          },
        },
      },
    },
    options: null,
  },
  function uniquenessTags(targetVal, _, ctx) {
    const duplicatedTags = getDuplicateTags(targetVal);
    if (duplicatedTags.length === 0) return;

    const results: IFunctionResult[] = [];

    for (const duplicatedIndex of duplicatedTags) {
      const duplicatedTag = targetVal[duplicatedIndex].name;

      results.push({
        message: `"tags" object contains duplicate tag name "${duplicatedTag}".`,
        path: [...ctx.path, duplicatedIndex, 'name'],
      });
    }

    return results;
  },
);
```

Reply:
@P0lip Yes, it is much easier, thank you very much! Suggestion applied.